src/python/pants/engine/unions.py

```diff
@@ -12,41 +12,22 @@
 
 @decorated_type_checkable
 def union(cls):
-    """A class decorator which other classes can specify that they can resolve to with `UnionRule`.
+    """A class decorator to allow a class to be a union base in the engine's mechanism for
+    polymorphism.
 
-    Annotating a class with @union allows other classes to use a UnionRule() instance to indicate
-    that they can be resolved to this base union class. This class will never be instantiated, and
-    should have no members -- it is used as a tag only, and will be replaced with whatever object is passed
-    in as the subject of a `await Get(...)`. See the following example:
+    Annotating a class with @union allows other classes to register a `UnionRule(BaseClass,
+    MemberClass)`. Then, you can use `await Get(Output, UnionBase, concrete_union_member)`. This
+    would be similar to writing `UnionRule(Output, ConcreteUnionMember,
+    concrete_union_member_instance)`, but allows you to write generic code without knowing at the
```

Comment:
This last sentence is a bit unclear. I might reword it as "... but allows you to write generic code without knowing what concrete classes might later implement that union."